### DISCUSSION TITLE
[mini.map] Issue with symbols in mini.map

### DISCUSSION THREAD

**Discussion by Jomik** [nvim-mini/mini.nvim#1166] [Q&A] [answered]
I have an issue where the symbols in the minimap show up like unknown characters. They look just fine in the normal buffers. I am using the dot encoding
![image](https://github.com/user-attachments/assets/0e22c07e-88c7-427f-832b-5b86ea3ec8d0)
Any ideas what causes this?

```lua

    "echasnovski/mini.map",
    version = "*",
    opts = function(_, opts)
      local mini_map = require("mini.map")
      opts.integrations = {
        mini_map.gen_integration.builtin_search(),
        mini_map.gen_integration.diff(),
        mini_map.gen_integration.diagnostic(),
      }
      opts.symbols = {
        encode = mini_map.gen_encode_symbols.dot("4x2"),
      }
    end,
    init = function(_, opts)
      local mini_map = require("mini.map")
      mini_map.setup(opts)
      vim.api.nvim_create_autocmd("VimEnter", {
        callback = mini_map.open,
      })

      vim.api.nvim_create_autocmd("WinClosed", {
        callback = mini_map.refresh,
      })

      vim.api.nvim_create_autocmd({ "CursorHold", "CursorHoldI" }, {
        callback = function()
          local _, col = unpack(vim.api.nvim_win_get_cursor(0))
          local width = vim.api.nvim_win_get_width(0)
          if width - col < 20 then
            mini_map.close()
          else
            mini_map.open()
          end
        end,
      })
    end,
  }
```

**Comment by echasnovski** [comment]
That's the first time I am seeing this kind of issue (work in regular buffer but not in floating window). Here is what I'd suspect first:
- Positive 'winblend' which affects how characters are rendered in floating window. I.e. try to setting `window.winblend = 0` in `require('mini.map')` and see if it fixes the issue.
- General terminal emulator issue. Try different terminal emulator. By the way, what terminal emulator and OS is this from?

**Reply by Jomik** [reply]
I tried setting this in the above opts function for lazy.nvim, same issue.
```lua
      opts.window = {
        winblend = 0,
      }
```

This is MacOS, Neovide. Here is a help page with the dots visible.
![image](https://github.com/user-attachments/assets/933c88c5-ff24-4bb3-bcff-023032dd1225)
Similar issue in iTerm:
![image](https://github.com/user-attachments/assets/40d6a7e7-2dc1-4383-b838-73b1caa2b2e8)

**Comment by Jomik** [comment, marked as answer]
Sooo, I changed my config to move my `opts` into `init`, and that fixed it.
So I must have done something wrong ... My original config is in the top post.
```lua
{
    "echasnovski/mini.map",
    version = "*",
    init = function(_, opts)
      local mini_map = require("mini.map")
      opts = {
        integrations = {
          mini_map.gen_integration.builtin_search(),
          mini_map.gen_integration.diff(),
          mini_map.gen_integration.diagnostic(),
        },
        symbols = {
          encode = mini_map.gen_encode_symbols.dot("4x2"),
        },
      }
      mini_map.setup(opts)
      vim.api.nvim_create_autocmd("VimEnter", {
        callback = mini_map.open,
      })

      vim.api.nvim_create_autocmd("WinClosed", {
        callback = mini_map.refresh,
      })

      vim.api.nvim_create_autocmd({ "CursorHold", "CursorHoldI" }, {
        callback = function()
          local _, col = unpack(vim.api.nvim_win_get_cursor(0))
          local width = vim.api.nvim_win_get_width(0)
          if width - col < 20 then
            mini_map.close()
          else
            mini_map.open()
          end
        end,
      })
    end,
  }
  ```

**Reply by echasnovski** [reply]
Glad that it got resolved! Although definitely a strange issue.

**Reply by Jomik** [reply]
I think it may be because `.setup` is called twice? I realise that I should use `config` and not `init` since I call `setup` 🤔  At least my initial configuration also seems to work if I use `config` instead of `init`.

**Reply by echasnovski** [reply]
I don't think calling `setup()` twice should have that effect.

**Reply by Jomik** [reply]
No, I agree 🤔
